### PR TITLE
TS Clean up placeholder and add it to formfield updated docs

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR starts the clean up of the over use of type definitions 
This PR cleans up the placeholder
#### Where should the reviewer start?
js/utils/index.d.ts
#### What testing has been done on this PR?
manually tested by branching out to a typescript project
#### What are the relevant issues?
relevant issue #3237
#### Do the grommet docs need to be updated?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/FormField/doc.js [hunk 1]
```diff
@@ -42,6 +42,9 @@ export const doc = FormField => {
     pad: PropTypes.bool.description(
       'Whether to add padding to align with the padding of TextInput.',
     ),
+    placeholder: PropTypes.node.description(
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js]: `placeholder` isn't a prop of FormField and, should be removed from FormField associated props.

File: src/js/components/FormField/index.d.ts [hunk 2]
```diff
@@ -7,11 +8,12 @@ export interface FormFieldProps {
...
   name?: string;
   pad?: boolean;
+  placeholder?: PlaceHolderType
   required?: boolean;
   component?: any;
   validate?: {regexp?: object,message?: string} | ((...args: any[]) => any);
 }
 
-declare const FormField: React.ComponentClass<FormFieldProps & JSX.IntrinsicElements['input']>;
+declare const FormField: React.ComponentClass<TextInputProps & Omit<JSX.IntrinsicElements['input'], 'placeholder'>>;
```
Comment (ShimiSun) [inline src/js/components/FormField/index.d.ts]: great, I guess it solves the problem. does the line breaking necessary?

File: src/js/components/Select/doc.js [hunk 3]
```diff
@@ -134,11 +134,9 @@ export const doc = Select => {
       children callback in order to render anything based on the current item.`,
     ).isRequired,
     open: PropTypes.bool.description(`Control the state of the component.`),
-    placeholder: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.node,
-      PropTypes.element,
-    ]).description('Placeholder text to use when no value is provided.'),
+    placeholder: PropTypes.node.description(
```
Comment (ShimiSun) [inline src/js/components/Select/doc.js]: I understand the clean up of `string`, but why did we decide to go with node rater than with element as the type?

Comment (britt6612) [reply src/js/components/Select/doc.js]: the node includes both the string and the element

File: src/js/components/FormField/doc.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/FormField/doc.js]: okay so we just want to accept it for TS maybe we can discuss

File: src/js/components/Select/doc.js (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/Select/doc.js]: yea, but using simply `element` isn't enough?

Comment (britt6612) [reply src/js/components/Select/doc.js]: we totally can do element and string but just element will not cover string

Comment (ShimiSun) [reply src/js/components/Select/doc.js]: let's go with element & string rather than node, it will be more accurate.

Comment (britt6612) [reply src/js/components/Select/doc.js]: agree!

File: src/js/components/FormField/doc.js [hunk 4]
```diff
@@ -42,6 +42,10 @@ export const doc = FormField => {
     pad: PropTypes.bool.description(
       'Whether to add padding to align with the padding of TextInput.',
     ),
+    placeholder: PropTypes.oneOfType([
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js]: The `placeholder` shouldn't be a prop of FormField. it's a global attribute of HTML input. it shouldn't be placed on the FormField documentation.

File: src/js/components/Select/README.md [hunk 5]
```diff
@@ -334,11 +334,10 @@ boolean
 
 **placeholder**
 
-Placeholder text to use when no value is provided.
+Placeholder to use when no value is provided.
 
 ```
 string
```
Comment (ShimiSun) [inline src/js/components/Select/README.md]: thanks for cleaning up the node type, let's run TS tests on select to make sure there are no regressions

File: src/js/components/FormField/index.d.ts (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/FormField/index.d.ts]: This fix might be needed for Select.

**Comment by britt6612** [comment]
Select does not use the HTML ['input' ] so we do not need to omit placeholder for that I just cleaned up the placeholder to accept a JSX element and a string.

File: src/js/components/TextInput/doc.js [hunk 6]
```diff
@@ -71,9 +71,10 @@ The suggestion contains the object chosen from the supplied suggestions.`,
     onSuggestionsClose: PropTypes.func.description(
       'Function that will be called when the suggestions drop is closed.',
     ),
-    placeholder: PropTypes.node.description(
-      'Placeholder to use when no value is provided.',
-    ),
+    placeholder: PropTypes.oneOfType([
```
Comment (ShimiSun) [inline src/js/components/TextInput/doc.js]: let's make sure there is no regression for TextInput.

File: src/js/utils/index.d.ts [hunk 7]
```diff
@@ -38,6 +38,7 @@ export {isObject, deepFreeze, deepMerge, removeUndefined};
 export type A11yTitleType = string;
 export type AlignContentType = "start" | "center" | "end" | "between" | "around" | "stretch";
 export type AlignSelfType = "start" | "center" | "end" | "stretch";
+export type ColorType = string | {dark?: string,light?: string}
 export type GridAreaType = string;
 export type MarginType = "none" | "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | {bottom?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,horizontal?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,left?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,right?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,top?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,vertical?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string} | string;
-export type ColorType = string | {dark?: string,light?: string}
+export type PlaceHolderType = string | JSX.Element
```
Comment (ShimiSun) [inline src/js/utils/index.d.ts]:
```suggestion
export type PlaceHolderType = string | JSX.Element;
```

File: src/js/utils/index.d.ts [hunk 8]
```diff
@@ -38,6 +38,7 @@ export {isObject, deepFreeze, deepMerge, removeUndefined};
 export type A11yTitleType = string;
 export type AlignContentType = "start" | "center" | "end" | "between" | "around" | "stretch";
 export type AlignSelfType = "start" | "center" | "end" | "stretch";
+export type ColorType = string | {dark?: string,light?: string}
```
Comment (ShimiSun) [inline src/js/utils/index.d.ts]:
```suggestion
export type ColorType = string | {dark?: string,light?: string};
```

File: src/js/components/TextInput/doc.js (hunk 6 above)
Comment (britt6612) [reply src/js/components/TextInput/doc.js]: Yes I will triple check for select and Input 👍

File: src/js/components/TextInput/doc.js [hunk 9]
```diff
@@ -74,6 +74,7 @@ The suggestion contains the object chosen from the supplied suggestions.`,
     placeholder: PropTypes.oneOfType([
       PropTypes.string,
       PropTypes.element,
+      PropTypes.node,
```
Comment (ShimiSun) [inline src/js/components/TextInput/doc.js]: I don't think we should add `node` if it doesn't contribute to solving the issue.

**Comment by britt6612** [comment]
closes #3205

**Pull request merged**